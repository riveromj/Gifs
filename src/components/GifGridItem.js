import React from 'react'
import PropTypes from 'prop-types';


export const GifGridItem = ({title, url}) => {

    return (
        <div className="card animate__backInDown">
           <img className='gif' src={url} alt={title}/>
           <p>{title}</p>
        </div>
    )
}

GifGridItem.protoTypes={
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired
}