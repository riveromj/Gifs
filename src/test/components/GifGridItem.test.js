import React from 'react';
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas en GifGridItem', () => {
    const title="el titulo"
    const url="http//:local"
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('mostrar el componente correto ', () => {

        expect(wrapper).toMatchSnapshot();
        
    });
    
    test('debe tener un parrafo con el title', () => {
        
        const p =wrapper.find('p')
        expect(p.text().trim()).toBe(title)
    });

    test('la imagen debe tener url y alt', () => {
        
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('debe tener animate__backInDown ', () => {

        const div = wrapper.find('div')

        expect(div.prop('className').includes('animate__backInDown')).toBe(true)
        
    })
    
    
    
    
})
