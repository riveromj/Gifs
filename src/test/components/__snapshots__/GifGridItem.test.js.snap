// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem mostrar el componente correto  1`] = `
<div
  className="card animate__backInDown"
>
  <img
    alt="el titulo"
    className="gif"
    src="http//:local"
  />
  <p>
    el titulo
  </p>
</div>
`;
